src/jinjanator/customize.py

```diff
@@ -0,0 +1,152 @@
+        return {}
+
+    def extra_tests(self) -> Mapping[str, FunctionType]:
+        return {}
+
+    _IMPORTED_METHOD_NAMES: ClassVar = [
+        f.__name__
+        for f in (j2_environment_params, j2_environment, alter_context, extra_filters, extra_tests)
+    ]
+
+    @classmethod
+    def from_file(cls, filename: str) -> "CustomizationModule":
+        """Create Customize object"""
+        if filename is not None:
+            return cls(imp_load_source("customize-module", filename))
+        return cls(None)
+
+
+def import_functions(filename: str) -> Mapping[str, FunctionType]:
+    """Import functions from file, return as a dictionary"""
+    m = imp_load_source("imported-funcs", filename)
+    return {name: func for name, func in inspect.getmembers(m) if inspect.isfunction(func)}
+
+
+def register_filters(j2env: jinja2.Environment, filters: Mapping[str, FunctionType]) -> None:
+    """Register additional filters"""
+    j2env.filters.update(filters)
+
+
+def register_tests(j2env: jinja2.Environment, tests: Mapping[str, FunctionType]) -> None:
+    """Register additional tests"""
+    j2env.tests.update(tests)  # type: ignore[arg-type]
+
+
+def import_filters(renderer_env: jinja2.Environment, filename: str) -> None:
+    """Import filters from a file"""
+    register_filters(renderer_env, import_functions(filename))
+
+
+def import_tests(renderer_env: jinja2.Environment, filename: str) -> None:
+    """Import tests from a file"""
+    register_tests(renderer_env, import_functions(filename))
+
+
+def apply(
+    customize: CustomizationModule,
+    renderer_env: jinja2.Environment,
+    filters: list[str],
+    tests: list[str],
+) -> None:
+    """Apply customizations"""
+    customize.j2_environment(renderer_env)
+
+    for fname in filters:
+        import_filters(renderer_env, fname)
+
+    for fname in tests:
+        import_tests(renderer_env, fname)
+
+    register_filters(renderer_env, customize.extra_filters())
+
+    register_tests(renderer_env, customize.extra_tests())
+
+
+def add_args(parser: ArgumentParser) -> ArgumentParser:
+    """Add args to the parser"""
+    vargs_warning = (
+        "NOTE: Due to the way python implements options, if this is"
+        " the last option before the template filename, you need"
+        " to add a -- separator before the template filename"
+    )
+
+    parser.add_argument(
+        "--customize",
+        default=None,
+        metavar="python-file.py",
+        dest="customize",
+        help="A Python file that implements hooks to fine-tune the jinjanator behavior",
+    )
+
+    parser.add_argument(
+        "--filters",
+        nargs="+",
+        default=[],
+        metavar="python-file",
+        dest="filters",
+        help="Load custom Jinja2 filters from a Python file(s). "
+        f"All top-level functions are imported as filters. {vargs_warning}",
+    )
+
+    parser.add_argument(
+        "--tests",
+        nargs="+",
```

Comment:
Same as above for `filters`, although `j2cli` didn't support that but it should have!

Reply:
I am wiling to change it, but it will introduce a slight backwards incompatibility. I will leave it up to you to decide which way your prefer it.